File: Leetcode-Problems/Divide Array into Equal Pairs.cpp
class Solution
{
public:
    bool divideArray(vector<int> &nums)
    {
        unordered_map<int, int> freq;
        for (int i : nums)
        {
            freq[i]++;
        }
        for (auto i : freq)
        {
            if (i.second % 2 != 0)
                return false;
        }
        return true;
    }
};

File: Leetcode-Problems/Divide Array into Equal Pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for (int i : nums) {
            mp.put(i, mp.getOrDefault(i, 0) + 1);
        }
        for (int i : mp.values()) {
            if (i % 2 != 0)
                return false;
        }
        return true;
    }
}